utils/updateTestPlanReadme.pl

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env perl
+
+use strict;
+use warnings;
+
+use Pod::Usage;
+use Getopt::Long;
+use File::Basename;
+
+# command line options
+my $dirloc = '.'; # directory for serching test cases
+my $help;
+my $DEBUG = 0;
+
+# global variables
+my $tcCounter = 0;
+
+sub main {
+    GetOptions(
+        'help|?'  => \$help,
+        'dir|d=s' => \$dirloc,
+        'debug'   => \$DEBUG,
+    ) or pod2usage(2);
+
+    if ($help) {
+        pod2usage(1);
+        exit;
+    }
+    opendir(my $dir, $dirloc) || die "cannot open directory $dirloc";
+
+    my @files = readdir $dir;
+    @files = sort {$a cmp $b} @files;
+    foreach my $f (@files) {
+        print "$dirloc/$f\n" if $DEBUG;
+        next if $f =~ /^\./;
+
+        if ( -d "$dirloc/$f" and "$f" =~ /-TP/ ) {
+            tcList( "$dirloc/$f" );
+        }
+    }
+    print "No test cases found\nUse -d to specify directory\n" if !$tcCounter;
+}
+
+sub tcList
+{
+    my $tcDir = shift;
+
+    my $output = "## Test cases list\n\n";
+    $output .= "<!-- Table generated by script " . basename ($0) . " from Zonemaster/Zonemaster utils directory -->\n\n";
+    $output .= '|Test Case |Test Case Description|' . "\n";
+    $output .= '|:---------|:--------------------|' . "\n";
+
+    opendir(my $dir, $tcDir);
+    my @files = readdir $dir;
+    @files = sort {$a cmp $b} @files;
+    foreach my $f (@files) {
+        next if $f =~ /^\./;
+        next if $f eq "README.md";
+        unless ($f =~ /^[a-z]+[0-9]+\.md$/) {
+            warn "Skip file with unknown file name pattern: $f\n";
+            next;
+        }
+        $output .= tcName("$tcDir/$f");
+    }
+
+    if ( ! grep( /^README\.md$/, @files ) ) {
+        warn "No README.md file in folder $tcDir\n";
+        return;
+    }
+
+    writeReadme( "$tcDir/README.md", $output );
+}
+
+sub writeReadme
+{
+    my ( $tcPath, $tcTable ) = @_;
+
+    # slurp README content until pattern
+    my $content = "";
+    open( my $in, '<', $tcPath ) or die "Cannot open file $tcPath: $!";
+    while( <$in> ) {
+        last if $_ =~ /^## Test cases list$/;
```

Comment:
If my suggestion above is done, then this line must match the HTML comment instead.

Reply:
Not quite: I’ve found that it was necessary to match either `## Test cases list` or the line containing the HTML comment. The code now stops at the first of the two patterns.